core/solidus_core.gemspec

```diff
@@ -42,6 +42,7 @@ Gem::Specification.new do |s|
   s.add_dependency 'mini_magick', '~> 4.10'
   s.add_dependency 'monetize', '~> 1.8'
   s.add_dependency 'kt-paperclip', ['>= 6.3', '< 8']
+  s.add_dependency 'psych', ['>= 3.1.0', '< 5.0']
```

Comment:
@waiting-for-dev @tvdeyen I added this line in https://github.com/solidusio/solidus/pull/4451/commits/10da2e7ff02da6f45cf11d635e6d600b09767215. From that commit: 

The permitted_classes keyword argument was introduced to psych starting from psych 3.1.0. See https://github.com/ruby/psych/commit/682abf20f0ddbc8d83e816022953536dad80c05b.

Ruby 2.5.9 installs 3.0.2 by default.

This fixes errors like this when running the specs:

```
  1416) Taxation system integration tests selling from new york to new york an order with a download and a shipment has a shipment with additional tax of 0.40
        Failure/Error: preferences[name] = value

        ArgumentError:
          unknown keywords: permitted_classes, aliases
        # ./lib/spree/preferences/preferable_class_methods.rb:86:in `block in preference'
        # ./lib/spree/testing_support/factories/shipping_method_factory.rb:29:in `block (3 levels) in <top (required)>'
        # ./spec/models/spree/tax/taxation_integration_spec.rb:597:in `block (3 levels) in <top (required)>'
```